tests/lca/imagebasedupgrade/cnf/upgrade-talm/tests/e2e-upgrade-test.go

```diff
@@ -51,68 +50,32 @@ var _ = Describe(
-			})
-
-			By("Deleting prep cgu created on target hub cluster", func() {
-				err := cnfhelper.DeleteIbuTestCguOnTargetHub(cnfinittools.TargetHubAPIClient, tsparams.PrepCguName,
-					tsparams.IbuCguNamespace)
-				Expect(err).ToNot(HaveOccurred(), "Failed to delete prep cgu on target hub cluster")
-			})
-
-			By("Deleting upgrade cgu created on target hub cluster", func() {
-				err := cnfhelper.DeleteIbuTestCguOnTargetHub(cnfinittools.TargetHubAPIClient, tsparams.UpgradeCguName,
-					tsparams.IbuCguNamespace)
-				Expect(err).ToNot(HaveOccurred(), "Failed to delete upgrade cgu on target hub cluster")
-			})
-
-			By("Creating, enabling rollback CGU and waiting for CGU status to report completed", func() {
-				rollbackCguBuilder := cgu.NewCguBuilder(cnfinittools.TargetHubAPIClient,
-					tsparams.RollbackCguName, tsparams.IbuCguNamespace, 1).
-					WithCluster(tsparams.TargetSnoClusterName).
-					WithManagedPolicy(tsparams.RollbackPolicyName).
-					WithCanary(tsparams.TargetSnoClusterName)
-				rollbackCguBuilder.Definition.Spec.Enable = ptr.To(true)
-
-				rollbackCguBuilder, err := rollbackCguBuilder.Create()
-				Expect(err).ToNot(HaveOccurred(), "Failed to create rollback CGU.")
-
-				_, err = rollbackCguBuilder.WaitUntilComplete(35 * time.Minute)
-				Expect(err).ToNot(HaveOccurred(), "Rollback CGU did not complete in time.")
-			})
-
-			By("Creating, enabling ibu finalize CGU and waiting for CGU status to report completed", func() {
-				finalizeCguBuilder := cgu.NewCguBuilder(cnfinittools.TargetHubAPIClient,
-					tsparams.FinalizeCguName, tsparams.IbuCguNamespace, 1).
-					WithCluster(tsparams.TargetSnoClusterName).
-					WithManagedPolicy(tsparams.FinalizePolicyName).
-					WithCanary(tsparams.TargetSnoClusterName)
-				finalizeCguBuilder.Definition.Spec.Enable = ptr.To(true)
-
-				finalizeCguBuilder, err := finalizeCguBuilder.Create()
-				Expect(err).ToNot(HaveOccurred(), "Failed to create finalize CGU.")
-
-				_, err = ibu.WaitUntilStageComplete("Idle")
-				Expect(err).NotTo(HaveOccurred(), "error waiting for idle stage to complete")
-
-				_, err = finalizeCguBuilder.WaitUntilComplete(5 * time.Minute)
-				Expect(err).ToNot(HaveOccurred(), "Finalize CGU did not complete in time.")
-			})
-
-			By("Deleting rollback cgu created on target hub cluster", func() {
-				err := cnfhelper.DeleteIbuTestCguOnTargetHub(cnfinittools.TargetHubAPIClient, tsparams.RollbackCguName,
-					tsparams.IbuCguNamespace)
-				Expect(err).ToNot(HaveOccurred(), "Failed to delete rollback cgu on target hub cluster")
-			})
-
-			By("Deleting finalize cgu created on target hub cluster", func() {
-				err := cnfhelper.DeleteIbuTestCguOnTargetHub(cnfinittools.TargetHubAPIClient, tsparams.FinalizeCguName,
-					tsparams.IbuCguNamespace)
-				Expect(err).ToNot(HaveOccurred(), "Failed to delete finalize cgu on target hub cluster")
+			By("Deleting upgrade ibgu created on target hub cluster", func() {
+				upgradeIbguBuilder := ibgu.NewIbguBuilder(
```

Comment:
In AfterAll you build a new IBGU upgradeIbguBuilder in order to delete it.  I understand why.

[In my PR I declare ](https://github.com/openshift-kni/eco-gotests/pull/222/files#diff-dea03563f03fb7c1e3c01004644f8e797e837b4b4bf16073963091cfef0997a5R19) "var newIbguBuilder *ibgu.IbguBuilder"  in Describe so in the AfterAll I don't need to build it again, I can just newIbguBuilder.DeleteAndWait

Both work.. is one way better than the other?

Reply:
thanks Josh, I've updated the test to use the same approach as in your test for consistency.